tests/failpoint/db_failpoint_test.go

```diff
@@ -267,6 +270,97 @@ func TestIssue72(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestTx_Rollback_Freelist(t *testing.T) {
+	db := btesting.MustCreateDBWithOption(t, &bolt.Options{PageSize: 4096})
+
+	bucketName := []byte("data")
+
+	t.Log("Populate some data to have at least 5 leaf pages.")
+	var keys []string
+	err := db.Update(func(tx *bolt.Tx) error {
+		b, terr := tx.CreateBucket(bucketName)
+		if terr != nil {
+			return terr
+		}
+		for i := 0; i <= 10; i++ {
+			k := fmt.Sprintf("t1_k%02d", i)
+			keys = append(keys, k)
+
+			v := make([]byte, 1500)
+			if _, terr := crand.Read(v); terr != nil {
+				return terr
+			}
+
+			if terr := b.Put([]byte(k), v); terr != nil {
+				return terr
+			}
+		}
+		return nil
+	})
+	require.NoError(t, err)
+
+	t.Log("Remove some keys to have at least 3 more free pages.")
+	err = db.Update(func(tx *bolt.Tx) error {
+		b := tx.Bucket(bucketName)
+		for i := 0; i < 6; i++ {
+			if terr := b.Delete([]byte(keys[i])); terr != nil {
+				return terr
+			}
+		}
+		return nil
+	})
+	require.NoError(t, err)
```

Comment:
can you add an assertion to count the leafs and free pages here? just in case the numbers change in the future

Reply:
Added one more check below. We are good as long as the freelist ID isn't empty.

```
require.Greater(t, len(beforeFreelistPgids), 0)
```